cmd/provider/daemon.go

```diff
@@ -125,6 +125,7 @@ func daemonCommand(cctx *cli.Context) error {
 	eng, err := engine.New(
 		engine.WithDatastore(ds),
 		engine.WithDataTransfer(dt),
+		engine.WithDirectAnnounce(cfg.DirectAnnounce.URLs...),
```

Comment:
> Two announcement protocols are supported, and used depending on the publisher kind

No, the announcement protocol used is not determined by the publisher kind. If there is an advertisement publisher, then _all available_ announce protocols are used. [here](https://github.com/filecoin-project/index-provider/pull/252/files#diff-2e6c4934f63ff9b712c2c346b33036af4724adf70b0801fff9b74f71b37fcd89L205)

Both an HTTP publisher or a dtsync publisher are announced using both gossipsub and http. 

I think this is better left as `WithDirectAnnounce` since we may also want to send direct announce messages over libp2p in the future.

Reply:
After changing the implementation to use multiaddrs, and since there is no need to support anything other than HTTP for sending direct announce messages,I have decided to go back to using URLs instead, for the following reasons:
- Communicates (or at least suggests) to the user that an HTTP endpoint is expected.
- Easier for users to to construct URL path if needed
- Does not require conversion from multihash to URL, and avoids any error cases associated with that.